torch/utils/_typing_utils.py

```diff
@@ -0,0 +1,15 @@
+"""Miscellaneous utilities to aid with typing."""
+
+from typing import Optional, TypeVar
+
+# Helper to turn Optional[T] into T when we know None either isn't
+# possible or should trigger an exception.
+T = TypeVar("T")
+
+
+def not_none(obj: Optional[T]) -> T:
+    if obj is None:
+        raise RuntimeError(
```

Comment:
```suggestion
        raise TypeError(
```
Shouldn't this technically raise a TypeError? or a ValueError?